docs/basics/UsageWithReact.md

```diff
@@ -74,196 +72,281 @@ I see the following components (and their props) emerge from this brief:
 * **`Footer`** is a component where we let user change visible todo filter.
   - `filter: string` is the current filter: `'SHOW_ALL'`, `'SHOW_COMPLETED'` or `'SHOW_ACTIVE'`.
   - `onFilterChange(nextFilter: string)`: Callback to invoke when user chooses a different filter.
+* **`Link`** is a link with a callback.
+  - `onClick()` is a callback to invoke when link is clicked.
 
 These are all presentational components. They don’t know *where* the data comes from, or *how* to change it. They only render what’s given to them.
 
 If you migrate from Redux to something else, you’ll be able to keep all these components exactly the same. They have no dependency on Redux.
 
-Let’s write them! We don’t need to think about binding to Redux yet. You can just give them fake data while you experiment until they render correctly.
+We also have some container components that connects to Redux. The container components are responsible for getting and transforming state. Remember that changes result in a new state as state is immutable.
 
-## Presentational Components
+* **`AddTodo`** is an input field with a button.
+  - `onAddClick(text: string)` is a callback to invoke when a button is pressed.
 
-These are all normal React components, so we won’t stop to examine them in detail. Here they go:
+* **`FilterLink`** gets active link from Redux and pass it as props to the presentational component, Link.
 
-#### `components/AddTodo.js`
+* **`VisibleTodoList`** gets todos from Redux, filter the todos and pass them as props to the presentational component, TodoList.
 
-```js
-import React, { Component, PropTypes } from 'react'
-
-export default class AddTodo extends Component {
-  render() {
-    return (
-      <div>
-        <input type='text' ref='input' />
-        <button onClick={e => this.handleClick(e)}>
-          Add
-        </button>
-      </div>
-    )
-  }
+Let’s write the components! We begin with the presentational components, so we don’t need to think about binding to Redux yet.
```

Comment:
Overall it feels like this section lost its focus. It may help to add two subheaders: "Presentational Components" and "Container Components" (inside "Designing Component Hierarchy"). This would better separate these paragraphs and provide some visual rhythm.